### DISCUSSION TITLE
xdrip and Medtrum sensor

### DISCUSSION THREAD

**Discussion by jasutisr** [NightscoutFoundation/xDrip#3429] [General]
when xdrip will work with the Medtrum S9 EasySense insulin pump sensor?

**Comment by Navid200** [comment]
Do you currently use this sensor?

**Reply by jasutisr** [reply]
Not at the moment, but I'm thinking of trying it. I'm using dexome now.

**Reply by Navid200** [reply]
It's very unlikely to add support for a device if no one can provide samples or perform tests.

**Reply by AidasK** [reply]
I can I can! How can I help? 

Probably integration should be done via bluetooth I could provide some bluetooth sniffed log

**Reply by padiazg** [reply]
How may I help? I'm currently using the S9 sensor with the EasySense app

**Reply by BulliMauro** [reply]
How can I help? It would be really important to have xdrip for my son with the Medtrum s9 sensor.
Unfortunately the Medtrum alarms are really bad, and it also does not send the blood sugar to the watch.
I ask the programmers to develop this too... thank you very much

**Comment by adenis76** [comment]
My son currently testing this sensor. I am also interested to help with integration.

**Comment by adenis76** [comment]
https://easyview.medtrum.eu/v3/#/monitor
Medtrum have his own "nightscout". Can we use this to add in xdrip web follower? Maybe that will be easier solution.

**Reply by Navid200** [reply]
xDrip can already follow Nightscout.
Can you test to see if it works?

**Reply by adenis76** [reply]
It's not working. They page required login to see any data.
![Screenshot_20240609_130059_Chrome](https://github.com/NightscoutFoundation/xDrip/assets/40305361/ddf32dab-f82b-48d8-8bea-778d81548fa0)
![Screenshot_20240609_130118_Chrome~2](https://github.com/NightscoutFoundation/xDrip/assets/40305361/53116876-6fa6-4cbb-87bd-bafb96f4f134)

**Reply by Navid200** [reply]
Someone who is using the device should be able to create and account and upload data to it.

**Reply by adenis76** [reply]
I am currently uploading but I am interested can we use this medtrum service to add like data source in xdrip web follower.

**Reply by Navid200** [reply]
xDrip already has a data source for Nightscout.
This may be a language problem.  
Did you not say this was Nightscout?

**Reply by adenis76** [reply]
Sorry for my English. That is not what I meant to say. I won't to say that they have some kind of web page like nightscout and that is maybe easier solution to add medtrum in xdrip.

**Reply by Navid200** [reply]
I see,

Let me see what I can find out.

**Reply by Navid200** [reply]
But, even if we set that up, that will only work when user has access to the internet.

**Reply by T-o-b-i-a-s** [reply]
Hello @adenis76 and @Navid200 , I got one Medrum Nano Sensor for a test run and used AI to get the solution implemented that you proposed to get the Glucose values from the Easyview Service. You can have a look at the commit here: 
https://github.com/NightscoutFoundation/xDrip/compare/master...T-o-b-i-a-s:xDrip:medrum-follower
Additional information about the API was fed to the AI from https://github.com/rodecker/medtrum-tray and https://github.com/pachi81/GlucoDataHandler

As you mentioned, the glucose values are only available when you have an Internet connection, but it's better than nothing and similar to a Nightscout follower setup.

Would you be interested that I create a Pull request to add the functionality to your main branch?

The prerequisites to use the feature are as follows:
1. Create an additional account with a different e-mail address as a "caregiver" on the [Medtrum Easy View](https://easyview.medtrum.eu/v3/#/login/login) system
2. After loggin in as the caregiver, add a connection to the main patient account by entering the e-mail address of the primary patient account you use with your EasyPatch app
3. In the EasyPatch App, go to Settings -> Account Security and add the newly created caregiver account with the secondary e-mail address as a remote follower

In xDrip with the [patch](https://github.com/NightscoutFoundation/xDrip/compare/master...T-o-b-i-a-s:xDrip:medrum-follower) integrated, then select "Medtrum Follower" as the source and enter the mail address of your secondary caregiver account as "EasyView Username" and its password under "EasyView Password".

**Reply by Navid200** [reply]
@T-o-b-i-a-s This is something I have proposed for making the titles consistent with standard Google (Android) practices:
https://github.com/NightscoutFoundation/xDrip/discussions/4408 

We also need to talk about how/where you are adding any new menu items.

As far as copyright and AI, please have a look at this:
https://github.com/NightscoutFoundation/xDrip/blob/master/CONTRIBUTING.md#policy-on-ai-based-submissions-pull-requests

But, we can talk about all of those after you open a PR as well.
Yes, please open a PR.

**Reply by T-o-b-i-a-s** [reply]
PR #4663 opened. Happy to discuss the topics mentioned above in the PR. I already converted all visible text to sentence case, I hope everything is covered.

**Comment by adenis76** [comment]
Yes, that is correct.

**Reply by adenis76** [reply]
If you want I can send you my login data.

**Comment by Stealthwick** [comment]
I have hardware and can test this if needed (or send to someone more qualified!)

**Comment by KlaidasV** [comment]
> when xdrip will work with the Medtrum S9 EasySense insulin pump sensor?

Do i understand xorrectly that still xdrip not working with s9 sensor and nano pump? Any ideas when it will have integration?

**Reply by Navid200** [reply]
As far as I know, xDrip does not interact with any pump.

There is no way anyone can add support for any sensor without a sample.

We may be able to add a sensor to the companion app support, which will require you to run the official app in addition to xDrip if support is added.

**Reply by fedemotta** [reply]
Hi, I will use the nano pump CGM very soon. In Argentina they are selling as a mandatory combo with the pump. Can you guide me to understand how to develop this feature?

**Comment by Stealthwick** [comment]
I have this sensor. Companion mode is unlikely to be an option as the
readings are not displayed in a system notification.

**Comment by BulliMauro** [comment]
Hi, my son started the test a few days ago. For now it seems to be working
well, it stops the basal when it goes too low... I can't give you
instructions, it will have to be the doctors. The carbohydrate count is
needed... best wishes

Il giorno sab 6 lug 2024 alle ore 15:35 Federico Nicolás Motta <
***@***.***> ha scritto:

**Comment by schlau** [comment]
hi there
any news on the support for medtrum s9 for xdrip?

**Comment by Stealthwick** [comment]
S9 is being discontinued and replaced with S10 / S15. I do not expect
you'll find anyone willing to work on S9 at this point.
